internal/controller/etcdcluster_controller_test.go

```diff
@@ -110,12 +110,21 @@ var _ = Describe("EtcdCluster Controller", func() {
 			// check that Service is created
 			svc := &v1.Service{}
 			err = k8sClient.Get(ctx, typeNamespacedName, svc)
-			Expect(err).NotTo(HaveOccurred(), "cluster headless Service state should exist")
+			Expect(err).NotTo(HaveOccurred(), "cluster headless Service should exist")
 			Expect(svc.Spec.ClusterIP).To(Equal("None"), "cluster Service should be headless")
 			// check that StatefulSet is created
 			sts := &appsv1.StatefulSet{}
 			err = k8sClient.Get(ctx, typeNamespacedName, sts)
 			Expect(err).NotTo(HaveOccurred(), "cluster statefulset should exist")
+			// check that Service is created
+			svc = &v1.Service{}
+			clientSvcName := types.NamespacedName{
+				Namespace: typeNamespacedName.Namespace,
+				Name:      controllerReconciler.getClientServiceName(etcdcluster),
+			}
+			err = k8sClient.Get(ctx, clientSvcName, svc)
+			Expect(err).NotTo(HaveOccurred(), "cluster client Service should exist")
+			Expect(svc.Spec.ClusterIP).NotTo(Equal("None"), "cluster client Service should NOT be headless")
```

Comment:
Does it make sense to check correct ports and label selectors to pods?

Reply:
What about ports, I don't think so. It'd better to check in e2e tests

Writing tests for label selectors won't be good enough neither as we can only test the perfect case - all service label selector are included in pods` labels, but this does not protect us from collisions